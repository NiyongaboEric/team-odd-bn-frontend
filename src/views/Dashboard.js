import React from 'react';
import Sidebar from './Dashboard/sidebar/index';

const Dashboard = () => {
  return (
    <Sidebar>
      <p>Welcome to Barefoot</p>
      <p>This is content</p>
    </Sidebar>
  );
};

export default Dashboard;
