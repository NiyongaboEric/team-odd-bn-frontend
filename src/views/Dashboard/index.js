import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Proptypes from 'prop-types';
import signoutUser from '../../redux/actions/signoutAction';


export const Dashboard = (props) => {
  const logout = async () => {
    await props.signoutUser();
    props.history.push('/signin');
  };

  return (
    <>
      <p>Welcome to Barefoot</p>
      <Link to="#" onClick={logout} className="signout">Logout</Link>
    </>
  );
};

const mapStateToProps = (state) => ({
  isLoggedIn: state,
});

Dashboard.propTypes = {
  signoutUser: Proptypes.func.isRequired,
  history: Proptypes.shape({
    push: Proptypes.func,
  }),
};

Dashboard.defaultProps = {
  history: {
    push: () => {},
  },
};

export default connect(mapStateToProps, { signoutUser })(Dashboard);
