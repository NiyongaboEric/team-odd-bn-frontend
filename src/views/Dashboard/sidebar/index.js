import React, { Component } from 'react'
import { connect } from 'react-redux'
import TopSide from './TopSide'
import LeftSide from './LeftSide'
import RightSide from './RightSide'
import SideDrawer from '../../../components/Sidebar/SideDrawer/SideDrawer'
import Backdrop from '../../../components/Sidebar/Backdrop/Backdrop'
import profileActions from '../../../redux/actions/profileActions'
import NotificationPanel from './NotificationPanel'
import '../../../assets/css/notification.scss'

export class Sidebar extends Component {
  constructor(props) {
    super(props)
    this.state = {
      sideDrawerOpen: false,
    }
  }

  componentDidMount() {
    const { getCurrentUserinfo } = this.props
    getCurrentUserinfo(this.props.viewProfile)
  }

  drawerToggleClickHandler = () => {
    this.setState(prevState => ({
      sideDrawerOpen: !prevState.sideDrawerOpen,
    }))
  }

  backdropClickHandler = () => {
    this.setState({ sideDrawerOpen: false })
  }

  render() {
    let backdrop
    const { sideDrawerOpen } = this.state
    const { children } = this.props

    if (sideDrawerOpen) {
      backdrop = <Backdrop click={this.backdropClickHandler} />
    }

    return (
      <> 
        <TopSide drawerClickHandler={this.drawerToggleClickHandler} />
        <SideDrawer show={sideDrawerOpen} />
        <div className="both-side">
        <NotificationPanel display={this.state.display} />
          <LeftSide />
          <RightSide RightSideContent={children} />
        </div>
        {backdrop}
      </>
    )
  }
}

const mapStateToProps = state => {
  return {
    profile: state.viewProfile.profile,
    profileError: state.profileError,
    notification: state.notification,
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getCurrentUserinfo: () => dispatch(profileActions()),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Sidebar)
