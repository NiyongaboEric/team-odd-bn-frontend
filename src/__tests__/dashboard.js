import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import 'mock-local-storage';
import DashboardDefault from '../views/Dashboard/index';
import { Dashboard } from '../views/Dashboard/index';
import mockData from '../__mocks__/fileMock';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const { signoutProps } = mockData;

let initialState;

describe('Expect signout to be connected with redux and dashboard', () => {
  let wrapper2; let store2; let wrapper3;
  beforeEach(() => {
    initialState = {
      state: {
        signout: true,
        isLoggedin: {},
      },
      isSignout: true,
    };
    store2 = mockStore(initialState);
    wrapper2 = shallow(<DashboardDefault store={store2} {...signoutProps} />);
  });

  it('Expect to render without problem', () => {
    expect(wrapper2).toMatchSnapshot();
  });

  it('Expect to logout by clicking a button', async () => {
    wrapper3 = shallow(<Dashboard {...signoutProps} />);
    const signout = wrapper3.find('.signout');
    signout.simulate('click');
    expect(signout.text()).toBe('Logout');
  });
});
