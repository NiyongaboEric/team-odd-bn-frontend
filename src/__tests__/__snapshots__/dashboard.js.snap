// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect signout to be connected with redux and dashboard Expect to render without problem 1`] = `ShallowWrapper {}`;
